### PR TITLE
[Feature] [File Connector] Supports writing column names when the output type is file (CSV)

### LINKED ISSUE #5443: [Feature][File Connector] Supports writing column names when the output type is file (CSV)
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

We do not support adding column names to the first line when using File type Sink to generate files. If this feature is supported, the generated CSV file will be easier to read.

Here are some references:

In  `TextWriteStrategy.getOrCreateOutputStream` method the write will create a new file when can not get outputstream from `beingWrittenOutputStream`. I think it is more appropriate to obtain column names from `SeaTunnelRowType` and write them after creating a new file output stream.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zck573693104**
close #5443

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [x] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/TextWriteStrategy.java [hunk 1]
```diff
@@ -133,15 +133,21 @@ private FSDataOutputStream getOrCreateOutputStream(@NonNull String filePath) {
                         OutputStream out =
                                 lzo.createOutputStream(fileSystemUtils.getOutputStream(filePath));
                         fsDataOutputStream = new FSDataOutputStream(out, null);
+                        fsDataOutputStream.write(String.join(",", seaTunnelRowType.getFieldNames()).getBytes());
+                        fsDataOutputStream.write(System.lineSeparator().getBytes());
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/TextWriteStrategy.java]: Use `fsDataOutputStream.write(rowDelimiter.getBytes());` ?

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/TextWriteStrategy.java [hunk 2]
```diff
@@ -133,15 +133,21 @@ private FSDataOutputStream getOrCreateOutputStream(@NonNull String filePath) {
...
                         break;
                     case NONE:
                         fsDataOutputStream = fileSystemUtils.getOutputStream(filePath);
+                        fsDataOutputStream.write(String.join(",", seaTunnelRowType.getFieldNames()).getBytes());
+                        fsDataOutputStream.write(System.lineSeparator().getBytes());
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/TextWriteStrategy.java]: Same as above.

**Comment by EricJoy2048** [comment]
Please fix code style by run `sh tools/spotless_check/pre-commit.sh`

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/TextWriteStrategy.java (hunk 1 above)
Comment (zck573693104) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/TextWriteStrategy.java]: yeah,i will fix it

**Comment by zck573693104** [comment]
> Please fix code style by run `sh tools/spotless_check/pre-commit.sh`

i will fix it

File: docs/en/connector-v2/sink/LocalFile.md [hunk 3]
```diff
@@ -166,6 +167,10 @@ When File Format is Excel,The maximum number of data items that can be cached in
 
 Writer the sheet of the workbook
 
+### enable_header_write [boolean]
+
+false:dont write header,true:write header.
```
Comment (ruanwenjun) [inline docs/en/connector-v2/sink/LocalFile.md]: It seems you only implement this in text format, if so it's needed to mention this only support in text format.

Comment (zck573693104) [reply docs/en/connector-v2/sink/LocalFile.md]: yeah，i will fix it

Comment (zck573693104) [reply docs/en/connector-v2/sink/LocalFile.md]: support text and csv

**Comment by EricJoy2048** [comment]
Please fix CI error.

**Comment by zck573693104** [comment]
> Please fix CI error.



> Please fix CI error.

yeah. fix it ok.

File: docs/en/connector-v2/sink/LocalFile.md [hunk 4]
```diff
@@ -30,26 +30,27 @@ By default, we use 2PC commit to ensure `exactly-once`
...
+| batch_size                       | int     | no       | 1000000                                    |                                                                                         |
+| compress_codec                   | string  | no       | none                                       |                                                                                         |
+| common-options                   | object  | no       | -                                          |                                                                                         |
+| max_rows_in_memory               | int     | no       | -                                          | Only used when file_format_type is excel.                                               |
+| sheet_name                       | string  | no       | Sheet${Random number}                      | Only used when file_format_type is excel.                                               |
+| enable_header_write              | boolean | no       | false                                      | Only used when file_format_type is text,csv. false:dont write header,true:write header. |
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/LocalFile.md]:
```suggestion
| enable_header_write              | boolean | no       | false                                      | Only used when file_format_type is text,csv. false:don't write header,true:write header. |
```

File: docs/en/connector-v2/sink/LocalFile.md [hunk 5]
```diff
@@ -166,6 +167,10 @@ When File Format is Excel,The maximum number of data items that can be cached in
 
 Writer the sheet of the workbook
 
+### enable_header_write [boolean]
+
+Only used when file_format_type is text,csv.false:dont write header,true:write header.
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/LocalFile.md]:
```suggestion
Only used when file_format_type is text,csv.false:don't write header,true:write header.
```

File: docs/en/connector-v2/sink/LocalFile.md [hunk 6]
```diff
@@ -30,26 +30,27 @@ By default, we use 2PC commit to ensure `exactly-once`
...
-| compress_codec                   | string  | no       | none                                       |                                                           |
-| common-options                   | object  | no       | -                                          |                                                           |
-| max_rows_in_memory               | int     | no       | -                                          | Only used when file_format_type is excel.                 |
-| sheet_name                       | string  | no       | Sheet${Random number}                      | Only used when file_format_type is excel.                 |
+|               name               |  type   | required |               default value                |                                         remarks                                          |
+|----------------------------------|---------|----------|--------------------------------------------|------------------------------------------------------------------------------------------|
```
Comment (zhilinli123) [inline docs/en/connector-v2/sink/LocalFile.md]: Can you directly add documentation about Connector V2 for this?
reference: https://github.com/apache/seatunnel/blob/dev/docs/en/connector-v2/source/Mysql.md

**Review by EricJoy2048 — APPROVED** [review]

**Review by FlechazoW — APPROVED** [review]
Nice job!

**Review by lightzhao — APPROVED** [review]

**Review by liugddx — APPROVED** [review]

**Pull request merged**